### PR TITLE
driver: telink: gpio isr table offset fix

### PR DESCRIPTION
**Pull request by interfer**
Fixed incorrect offset in _sw_isr_table for Level 2 Interrupts.

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

### CODE REVIEW

**Comment by andy-liu-telink** [comment]
@interfer 
This change might affect all the existing drivers. I suggest you go through all our driver files (xxx_b91.c).
If the two APIs below are used in any of these files, 
- riscv_plic_irq_enable()
- riscv_plic_set_priority()
you need to make additional changes like below 

```c
	irq_num -= CONFIG_2ND_LVL_ISR_TBL_OFFSET;
        ...
	riscv_plic_irq_enable(irq_num);
	riscv_plic_set_priority(irq_num, irq_prioriy);
```

I'm sure you need to modify these two files
- uart_b91.c
- ieeee802154_b91.c
You probably want to double check other files as well.

**Review by andy-liu-telink — APPROVED** [review]

File: boards/riscv/tlsr9518adk80d/tlsr9518adk80d_defconfig [hunk 1]
```diff
@@ -13,6 +13,7 @@ CONFIG_UART_INTERRUPT_DRIVEN=y
 CONFIG_SERIAL=y
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
+CONFIG_2ND_LVL_ISR_TBL_OFFSET=12
```
Comment (andriy-bilynskyy) [inline boards/riscv/tlsr9518adk80d/tlsr9518adk80d_defconfig]: Move to SOC

**Comment by andriy-bilynskyy** [comment]
driver: telink: gpio isr table offset fix #14 
driver: telink: ISR table offset fix

**Review by andriy-bilynskyy — CHANGES_REQUESTED** [review]

File: boards/riscv/tlsr9518adk80d/tlsr9518adk80d_defconfig (hunk 1 above)
Comment (interfer) [reply boards/riscv/tlsr9518adk80d/tlsr9518adk80d_defconfig]: removed from here, already defined in `zephyr/soc/riscv/riscv-privilege/telink_b91/Kconfig.defconfig.series`

File: drivers/serial/uart_b91.c [hunk 2]
```diff
@@ -583,41 +583,34 @@ static const struct uart_driver_api uart_b91_driver_api = {
...
-										    \
-	static void uart_b91_irq_connect_##n(void)				    \
-	{									    \
-		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority),		    \
-			    uart_b91_irq_handler,				    \
-			    DEVICE_DT_INST_GET(n), 0);				    \
-										    \
-		riscv_plic_irq_enable(DT_INST_IRQN(n));				    \
-		riscv_plic_set_priority(DT_INST_IRQN(n), DT_INST_IRQ(n, priority)); \
+#define UART_B91_INIT(n)                                                                           \
```
Comment (andriy-bilynskyy) [inline drivers/serial/uart_b91.c]: Do not change tabulation

Comment (interfer) [reply drivers/serial/uart_b91.c]: Done

**Review by andriy-bilynskyy — APPROVED** [review]

**Pull request merged**